client/src/app/pages/external/jira/instance-form.tsx

```diff
@@ -0,0 +1,292 @@
+    },
+    {
+      value: "jira-server",
+      toString: () => "Jira Server",
+    },
+    {
+      value: "jira-datacenter",
+      toString: () => "Jira Datacenter",
+    },
+  ];
+
+  const identityOptions = identities
+    .filter((identity) => identity.kind === "jira")
+    .map((identity) => {
+      return {
+        value: identity.name,
+        toString: () => identity.name,
+      };
+    });
+
+  return (
+    <Form onSubmit={handleSubmit(onSubmit)}>
+      {axiosError && (
+        <Alert
+          variant="danger"
+          isInline
+          title={getAxiosErrorMessage(axiosError)}
+        />
+      )}
+      <HookFormPFTextInput
+        control={control}
+        name="name"
+        label="Instance name"
+        fieldId="name"
+        isRequired
+      />
+      <HookFormPFTextInput
+        control={control}
+        name="url"
+        label="URL"
+        fieldId="url"
+        isRequired
+      />
+
+      <HookFormPFGroupController
+        control={control}
+        name="kind"
+        label="Instance type"
+        fieldId="type-select"
+        isRequired
+        renderInput={({ field: { value, name } }) => (
+          <SimpleSelect
+            id="type-select"
+            toggleId="type-select-toggle"
+            placeholderText={t("composed.selectMany", {
+              what: t("terms.instanceType").toLowerCase(),
+            })}
+            toggleAriaLabel="Type select dropdown toggle"
+            aria-label={name}
+            value={value ? toOptionLike(value, kindOptions) : undefined}
+            options={kindOptions}
+            onChange={(selection) => {
+              const selectionValue = selection as OptionWithValue<JiraKind>;
+              setValue(name, selectionValue.value);
+            }}
+          />
+        )}
+      />
+      <HookFormPFGroupController
+        control={control}
+        name="credentialName"
+        label={t("terms.credentials")}
+        fieldId="Credentials"
+        isRequired
+        renderInput={({ field: { value, name, onChange } }) => (
+          <SimpleSelect
+            id="credentials-select"
+            toggleId="credentials-toggle"
+            placeholderText={t("composed.selectMany", {
+              what: t("terms.credentials").toLowerCase(),
+            })}
+            toggleAriaLabel="Credentials select dropdown toggle"
+            aria-label={name}
+            value={value ? toOptionLike(value, identityOptions) : undefined}
+            options={identityOptions}
+            onChange={(selection) => {
+              const selectionValue = selection as OptionWithValue<string>;
+              onChange(selectionValue.value);
+            }}
+            onClear={() => onChange([])}
+          />
+        )}
+      />
+
+      <Switch
+        id="insecureCommunication"
+        className="repo"
+        label="Enable insecure communication"
+        aria-label="Insecure Communication"
+        isChecked={insecureCommunication}
```

Comment:
Looking great, Gilles. I am wondering if this one is just left out of the form state due to lack of API support?